server/src/main/java/org/apache/iotdb/db/metadata/idtable/AppendOnlyDiskSchemaManager.java

```diff
@@ -190,6 +198,65 @@ public Collection<DiskSchemaEntry> getAllSchemaEntry() throws IOException {
     return res;
   }
 
+  /**
+   * get DiskSchemaEntries from disk file
+   *
+   * @param offsets the offset of each record on the disk file
+   * @return DiskSchemaEntries
+   */
+  @Override
+  public List<DiskSchemaEntry> getDiskSchemaEntriesByOffset(List<Long> offsets) {
+    List<DiskSchemaEntry> diskSchemaEntries = new ArrayList<>(offsets.size());
+    Collections.sort(offsets);
+    try (RandomAccessFile randomAccessFile = new RandomAccessFile(dataFile, "r")) {
+      for (long offset : offsets) {
```

Comment:
Make this an attribute of this class. Frequent open operation of file will cost serious performance loss.

Reply:
fixed